File: leetcode/两数和.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    const n = nums.length;
    for(let i = 0; i<n;i++){
        for(let j = i +1;j<n;j++){
            if(nums[i] + nums[j] ==target){
                return [i,j];
            }
        }
    }
    return[];
};

var twoSum = function(nums, target) {
    const seen = new Map();
    for(let i = 0;i<nums.length;i++){
        const num = nums[i];
        const need = target - num;
        if(seen.has(need)){
            return [seen.has(need),i];
        }
        seen.set(num,i);
    }
    return [];
}

File: leetcode/两数和.py
from typing import List
# 暴力解决
class Solution:
    def twoSum(self,nums:List[int],target:int) ->List[int]:
        for i in range(len(nums)):
            for j in range(i+1 ,len(nums)):
                if nums[i]+nums[j] == target:
                    return [i,j]

# 哈希表
class Solution:
    def twoSum(self,nums:List[int],target:int) ->List[int]:
        seen = {}
        for i,j in enumerate(nums):
            need = target - j
            if need in seen:
                return [seen[need],i]
            seen[j] = i